library/public/interactor.h

```diff
@@ -65,29 +87,40 @@ class F3D_EXPORT interactor
    *
    * modifiers is a binary flag from the dedicated enum that represent KeyModifiers.
    *
-   * Adding commands for an existing combination of interaction and modifier will replace it.
-   *
    * When the corresponding interaction and modifiers happens, the provided commands will be
    * triggered using triggerCommand.
+   * Considering checkinng if an interaction exists or removing it before adding it to avoid potential conflicts.
+   *
    * ANY modifier interactions will only be triggered if no other interaction bind with modifier
    * is found.
+   *
+   * Adding commands for an existing combination of interaction and modifier will throw a
+   * interactor::already_exists_exception.
    */
   virtual interactor& addInteractionCommands(
-    std::string interaction, ModifierKeys modifiers, std::vector<std::string> commands) = 0;
+    const std::string& interaction, ModifierKeys modifiers, std::vector<std::string> commands) = 0;
 
   /**
    * See addInteractionCommands
    * Convenience method to add a single command for an interaction, similar as
    * addInteractionCommands(interaction, modifiers, {command})
+   *
+   * Adding command for an existing combination of interaction and modifier will throw a
+   * interactor::already_exists_exception.
    */
   virtual interactor& addInteractionCommand(
-    std::string interaction, ModifierKeys modifiers, std::string command) = 0;
+    const std::string& interaction, ModifierKeys modifiers, std::string command) = 0;
 
   /**
    * Remove interaction commands corresponding to provided interaction and modifiers
    */
   virtual interactor& removeInteractionCommands(
     std::string interaction, ModifierKeys modifiers) = 0;
+
+  /**
+   * Return a string vector of all currently defined interaction binds
+   */
+  virtual std::vector<std::pair<std::string, ModifierKeys>> getInteractionBinds() = 0;
```

Comment:
not sure about naming here, it could be

` getInteractionCommandsBinds` instead.